src/Analyzer.BicepProcessor/BicepTemplateProcessor.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Microsoft.Azure.Templates.Analyzer.TemplateProcessor;
+using System.IO;
+using Bicep.Core.Emit;
+using Bicep.Core.FileSystem;
+using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+using Bicep.Core.Workspaces;
+using Bicep.Core.TypeSystem.Az;
+
+namespace Microsoft.Azure.Templates.Analyzer.BicepProcessor
+{
+    public class BicepTemplateProcessor
+    {
+
+        private readonly string bicep;
+        private readonly string bicepPath;
+        private readonly string apiVersion;
+        private readonly bool dropResourceCopies;
+
+        /// <summary>
+        ///  Constructor for the ARM Template Processing library
+        /// </summary>
+        /// <param name="bicepTemplate">The ARM Template <c>JSON</c>. Must follow this schema: https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#</param>
+        /// <param name="bicepPath">The ARM Template <c>JSON</c>. Must follow this schema: https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#</param>
+        /// <param name="apiVersion">The deployment API version. Must be a valid version from the deploymetns list here: https://docs.microsoft.com/en-us/azure/templates/microsoft.resources/allversions</param>
+        /// <param name="dropResourceCopies">Whether copies of resources (when using the copy element in the ARM Template) should be dropped after processing.</param>
+        public BicepTemplateProcessor(string bicepTemplate, string bicepPath, string apiVersion = "2020-01-01", bool dropResourceCopies = false)
+        {
+            this.bicep = bicepTemplate;
+            this.bicepPath = bicepPath;
+            this.apiVersion = apiVersion;
+            this.dropResourceCopies = dropResourceCopies;
+        }
+        public string ConvertBicepToJson()
+        {
+            using var stringWriter = new StringWriter();
+            var syntaxTreeGrouping = SyntaxTreeGroupingBuilder.Build(new FileResolver(), new Workspace(), PathHelper.FilePathToFileUrl(bicepPath));
+            var compilation = new Compilation(AzResourceTypeProvider.CreateWithAzTypes(), syntaxTreeGrouping);
+            var emitter = new TemplateEmitter(compilation.GetEntrypointSemanticModel(), "0.4.123.62267");
```

Comment:
Should this assembly file version be able to be passed through the constructor or will it never change? Is it defined as an enum or something in the Bicep library?

Reply:
@anthony-c-martin Does the `assemblyFileVersion` parameter for `TemplateEmitter` matter for our context? Can we set it to `null` or an empty string? I don't want to have to update this value in the future if it's not necessary.

Reply:
You can set it to empty string if you like - it's only used to stamp metadata into the generated JSON for telemetry purposes (example [here](https://github.com/Azure/bicep/blob/f1a4461a84bdb7b3c0f5e329b64387b0ba73fe72/docs/examples/101/1vm-2nics-2subnets-1vnet/main.json#L7)).